From f34b350a65c311e1521b36446f70a632d53d6317 Mon Sep 17 00:00:00 2001
From: David Vrabel <david.vrabel@citrix.com>
Date: Thu, 18 Apr 2013 17:22:15 +0100
Subject: [PATCH 09/10] x86: check kexec relocation code fits in a page

The kexec relocation (control) code must fit in a single page so add a
link time check for this.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
---
 xen/arch/x86/xen.lds.S |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xen/arch/x86/xen.lds.S b/xen/arch/x86/xen.lds.S
index d959941..eebed01 100644
--- a/xen/arch/x86/xen.lds.S
+++ b/xen/arch/x86/xen.lds.S
@@ -186,3 +186,7 @@ SECTIONS
   .stab.indexstr 0 : { *(.stab.indexstr) }
   .comment 0 : { *(.comment) }
 }
+
+#ifdef CONFIG_KEXEC
+ASSERT(__kexec_reloc_size <= PAGE_SIZE, "kexec control code is too large")
+#endif
-- 
1.7.10.4

